movescu/src/main.rs

```diff
@@ -0,0 +1,540 @@
+                    }
+
+                    // fetch DICOM data
+                    let dcm = if let Some(second_pdata) = data.get(1) {
+                        InMemDicomObject::read_dataset_with_ts(second_pdata.data.as_slice(), ts)
+                            .whatever_context("Could not read response data set")?
+                    } else {
+                        let mut rsp = scu.receive_pdata();
+                        let mut response_data = Vec::new();
+                        rsp.read_to_end(&mut response_data)
+                            .whatever_context("Failed to read response data")?;
+
+                        InMemDicomObject::read_dataset_with_ts(&response_data[..], ts)
+                            .whatever_context("Could not read response data set")?
+                    };
+
+                    println!(
+                        "------------------------ Match #{i} ------------------------"
+                    );
+                    DumpOptions::new()
+                        .dump_object(&dcm)
+                        .context(DumpOutputSnafu)?;
+
+                    // check DICOM status in response data,
+                    // as some implementations might report status code 0
+                    // upon sending the response data
+                    if let Some(status) = dcm.get(tags::STATUS) {
+                        let status = status.to_int::<u16>().ok();
+                        if status == Some(0) {
+                            if verbose {
+                                debug!("Matching is complete");
+                            }
+                            break;
+                        }
+                    }
+
+                    i += 1;
+                } else {
+                    let msg = format!("Operation failed (status code {status:x})");
+
+                    if status == 0xa701 {
+                        warn!("{} Out of resources (number of matches)", msg);
+                    } else if status == 0xa702 {
+                        warn!("{} Out of resources (sub-operations)", msg);
+                    } else if status == 0x0122 {
+                        warn!("{} SOP class not supported", msg);
+                    } else if status == 0xa801 {
+                        warn!("{} Move destination unknown", msg);
+                    } else if status == 0xa900 {
+                        warn!("{} Identifier does not match SOP class in C-MOVE response", msg);
+                    } else if status == 0xc000 {
+                        warn!("{} Unable to process C-MOVE response", msg);
+                    } else if status == 0xfe00 {
+                        warn!("{} Sub-operations terminated due to cancel indication", msg);
+                    } else if status == 0xb000 {
+                        warn!("{} Sub-operations complete with one or more failures", msg);
+                    } else {
+                        warn!("{} Unknown status code", msg);
+                    }
```

Comment:
Not very important, but this would be cleaner and involve even less allocations.
```suggestion
                    let msg = match status {
                        0xa701 => "Out of resources (number of matches)",
                        0xa702 => "Out of resources (sub-operations)",
                        0x0122 => "SOP class not supported",
                        0xa801 => "Move destination unknown",
                        0xa900 => "Identifier does not match SOP class in C-MOVE response",
                        0xc000 => "Unable to process C-MOVE response",
                        0xfe00 => "Sub-operations terminated due to cancel indication",
                        0xb000 => "Sub-operations complete with one or more failures",
                        _ => "Unknown status code",
                    };
                    warn!("Operation failed (status code {status:x}) {msg}");
```